cdap-ui/app/directives/widget-container/widget-sql-conditions/widget-sql-conditions.js

```diff
@@ -23,8 +23,32 @@ function SqlConditionsController() {
 
   vm.mapInputSchema = {};
   vm.stageList = [];
-
+  vm.error = null;
+
+  /**
+   * FIXME: CDAP-14999
+   * This exists as a stopgap solution when user clicks on generate schema on joiner plugin
+   * where the input stages have either & or . or = in their name.
+   * The right fix is to fix it in backend.
+   */
+  vm.checkRulesForValidStageNames = () => {
+    const invalidRule = /[&\.=]/g;
+    let invalidStageNames = [];
+    angular.forEach(vm.rules, (rule) => {
+      var invalidStageName = rule.filter(field => invalidRule.test(field.stageName));
+      if (invalidStageName.length) {
+        invalidStageNames = invalidStageNames.concat(invalidStageName.map(stage => stage.stageName));
+      }
+    });
+    if (invalidStageNames.length) {
+      vm.error = 'Invalid name for ' + (invalidStageNames.length > 1 ? 'nodes' : 'node') + ': ' +
```

Comment:
Since this message is popping up on the joiner plugin and not the node with the invalid name, we could help the user find the node that needs to be changed by saying "input node" or "input nodes" in the error message.

Reply:
FYI @bdmogal Adding "input node(s)" to the above message.